docs/sources/next/javascript-api/k6-experimental/browser/browsercontext/cookies.md

```diff
@@ -5,17 +5,17 @@ description: 'Retrieves context cookies.'
 
 # cookies([urls])
 
-Returns a list of [cookies](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/browser/browsercontext/cookie) from the [BrowserContext](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/browser/browsercontext) filtered by the provided `urls`. If no `urls` are provided, all cookies are returned.
+Returns a list of [cookies](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/browser/browsercontext/cookie) from the [browser context](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/browser/browsercontext) filtered by the provided `urls`. If no `urls` are provided, all cookies are returned.
 
 | Parameter | Type  | Description                                                                                                                                                                                                                                                                    |
 | --------- | ----- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
-| urls      | array | A string array of URLs to filter the [cookies](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/browser/browsercontext/cookie) in the [BrowserContext](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/browser/browsercontext). |
+| urls      | array | A string array of URLs to filter the [cookies](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/browser/browsercontext/cookie) in the [browser context](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/browser/browsercontext). |
 
 ### Returns
 
-| Type  | Description                                                                                                                 |
-| ----- | --------------------------------------------------------------------------------------------------------------------------- |
-| array | A list of [cookies](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/browser/browsercontext/cookie). |
+| Type                     | Description                                                                                                                                                |
+| ------------------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `Promise<Array<Cookie>>` | A Promise that fulfills with an array of [cookies](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/browser/browsercontext/cookie). |
```

Comment:
It is common to use simple array types in typescript, i.e. `Cookie[]`, for simple types. The generic version is usually preserved for more complex types, e.g. `Array<typeof someVar>`, `Array<MyGeneric<string>>`.

TLDR; I think we should use `Promise<Inner[]>` as much as we can, because it's easier to read.

Reply:
Good point 👍 I follow PW's approach to be consistent with them. As people usually wonder how k6-browser is different compared to PW's. WDYT?

Reply:
> - How is k6-browser different from PW?
> - Our docs have better type signatures

Personally, I think the (arguably) improved readability outweighs being consistent with PW's docs. It also appears to be convention we've used in the type defs (although there aren't a lot of array types).

That said, it's not a hill I'm willing to die on.

Reply:
Created an issue to do this later #1593